### PR TITLE
logtail: change batched upload mechanism to not use CPU when idle

### PR DESCRIPTION
**Pull request by bradfitz**
The mobile implementation had a 2 minute ticker going all the time to do a channel send. Instead, schedule it as needed based on activity.

Then we can be actually idle for long periods of time.

Updates #3363

### CODE REVIEW

File: logtail/logtail.go [hunk 1]
```diff
@@ -247,18 +249,10 @@ func (l *Logger) Close() {
 // If the caller provides a DrainLogs channel, then unblock-drain-on-Write
```
Comment (nickoneill) [inline logtail/logtail.go:249]: `drainWake` now

**Review by nickoneill — APPROVED** [review]

**Review by maisem — APPROVED** [review]

**Comment by dsnet** [comment]
Wow... so a 2 minute ticker makes a big difference in terms of battery usage?

**Comment by bradfitz** [comment]
> Wow... so a 2 minute ticker makes a big difference in terms of battery usage?

It at least shows up I guess? cc @nickoneill @mihaip

**Comment by dsnet** [comment]
The data-plane audit logging feature uses a 5-second ticker 😨.

I'll refactor it in the future to be activity based as well.

**Comment by nickoneill** [comment]
I wouldn't call it a significant impact, but it did seem to make a visual difference in the usage chart when I set the timer to 6h rather than 2m 🙃

File: logtail/logtail.go [hunk 2]
```diff
@@ -241,24 +253,16 @@ func (l *Logger) Close() {
...
 // drainBlock is called by drainPending when there are no logs to drain.
 //
-// In typical operation, every call to the Write method unblocks and triggers
-// a buffer.TryReadline, so logs are written with very low latency.
+// In typical operation, every call to the Write method unblocks and triggers a
+// buffer.TryReadline, so logs are written with very low latency.
 //
-// If the caller provides a DrainLogs channel, then unblock-drain-on-Write
-// is disabled, and it is up to the caller to trigger unblock the drain.
+// If the caller specified FlushInterface, drainWake is only sent to
```
Comment (mihaip) [inline logtail/logtail.go:259]: Should this have been `FlushDelay`?

**Pull request merged**